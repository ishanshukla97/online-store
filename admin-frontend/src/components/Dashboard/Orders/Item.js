import React from "react";
import { connect } from "react-redux";
import { ORDER_STATUS } from "../../../utils/constants";
import "./index.css";

const Item = props => {
	const [isOpen, setIsOpen] = React.useState(false);
	console.log(props);

	const productDetails = () => {
		return props.products.map(product => {
			return (
				<div key={product._id}>
					<span style={{ paddingRight: "3px" }}>{product.name}</span>
					<span style={{ fontWeight: "600" }}>
						QTY: x{product.quantity}
					</span>
				</div>
			);
		});
	};

	const setStatus = status => {
		props.setStatus({
			orderId: props._id,
			status
		});
	};

	const orderActions = () => {
		switch (props.status) {
			case "PENDING":
				//render confirm and cancel btn
				return (
					<div>
						<div
							className="btn-small confirmed"
							onClick={() => setStatus(ORDER_STATUS.CONFIRMED)}
						>
							CONFIRM
						</div>
						<div
							className="btn-small cancel"
							onClick={() => setStatus(ORDER_STATUS.CANCELED)}
						>
							CANCEL
						</div>
					</div>
				);

			case "CONFIRMED":
				//render delivered btn
				return (
					<div>
						<div
							className="btn-small delivered"
							onClick={() => setStatus(ORDER_STATUS.DELIVERED)}
						>
							DELIVERED
						</div>
						<div
							className="btn-small cancel"
							onClick={() => setStatus(ORDER_STATUS.CANCELED)}
						>
							CANCEL
						</div>
					</div>
				);
			default:
				return;
		}
	};

	return (
		<li className="expandable-item-wrapper">
			<div
				className={"expandable-header " + props.status.toLowerCase()}
				onClick={() => setIsOpen(!isOpen)}
			>
				<div className="subsection-left">{props.creator.name}</div>
				<div className="subsection-center">{props.creator.contact}</div>
				<div className="subsection-right">{props.total}</div>
			</div>
			{isOpen ? (
				<div
					className={"expandable-body " + props.status.toLowerCase()}
				>
					<div className="subsection-left-body">
						{productDetails()}
					</div>
					<div className="subsection-center-body">
						{props.creator.address}
					</div>
					<div className="subsection-right-body">
						{orderActions()}
					</div>
				</div>
			) : (
				<div />
			)}
		</li>
	);
};

const mapStateToProps = ({ products }) => {
	return { allProducts: products };
};

export default connect(mapStateToProps)(Item);
