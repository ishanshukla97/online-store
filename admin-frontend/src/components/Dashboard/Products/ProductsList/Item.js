import React from "react";

const Item = props => {
	return (
		<a href="#!" className="collection-item" onClick={props.onClick}>
			{props.item.title}
		</a>
	);
};

export default Item;
