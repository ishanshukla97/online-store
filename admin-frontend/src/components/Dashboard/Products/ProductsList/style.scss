@import "../../../../sass/main";

.products-table {
	position: relative;
	width: 80%;
	margin: 1rem;
	@include box-shadow-1;
	border-radius: 5px;
	padding: 4rem;
	background-color: $color-grey-light-1;
	font-size: 1.5rem;

	&__item {
		cursor: pointer;
		transition: all 0.2s;

		&:hover {
			background-color: $color-grey-light-2;
		}
		&:active {
			background-color: $color-grey-dark-2;
		}
	}
}
