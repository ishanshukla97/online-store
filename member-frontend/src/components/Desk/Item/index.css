.card {
	animation-name: fadein;
	animation-duration: 300ms;
	animation-timing-function: ease-in;
}

@keyframes fadein {
	0% {
		opacity: 0;
		top: 4px;
	}

	100% {
		opacity: 1;
		top: 0px;
	}
}

.card:hover {
	box-shadow: 0px 5px 20px 3px rgba(0, 0, 0, 0.5) !important;
}

.add-btn {
	float: right;
}

.price-text {
	font-size: 14pt;
}
