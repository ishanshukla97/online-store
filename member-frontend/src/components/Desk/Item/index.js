import React from "react";
import "./index.css";
const Item = props => {
	return (
		<div className="col m4" key={props._id}>
			<div className="card medium">
				<div className="card-image">
					<img src="https://images.unsplash.com/photo-1561363702-e07252da3399?ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80" />
					<span className="card-title">{props.title}</span>
				</div>
				<div className="card-content">
					<p>{props.description}</p>
				</div>
				<div className="card-action">
					<a href="#!" className="price-text">
						â‚¹ {props.price}
					</a>
					<a
						href="#!"
						className="add-btn btn-floating red waves-effect"
						onClick={() => {
							console.log(props.title);
						}}
					>
						<i className="material-icons">add</i>
					</a>
				</div>
			</div>
		</div>
	);
};

export default Item;
