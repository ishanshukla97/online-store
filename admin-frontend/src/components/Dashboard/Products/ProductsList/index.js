import React from "react";
import Item from "./Item";
import { Table } from "react-bootstrap";
import "./style.scss";

const ProductsList = props => {
	if (props.products.length < 1) {
		return <div>Loading...</div>;
	}
	return (
		<Table className="products-table" responsive="sm">
			<thead className="products-table__head">
				<tr>
					<th>ID</th>
					<th>Name</th>
					<th>Category</th>
					<th>Price</th>
				</tr>
			</thead>
			<tbody>
				{props.products.map(product => {
					return (
						<Item
							key={product._id}
							item={product}
							onClick={() => props.onClick(product._id)}
						/>
					);
				})}
			</tbody>
		</Table>
	);
};
export default ProductsList;
