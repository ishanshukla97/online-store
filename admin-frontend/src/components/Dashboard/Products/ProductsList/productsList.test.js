import React from "react";
import renderer from "react-test-renderer";
import Item from "./Item";
import ProductsList from "./index";

const mockItemList = [
	{
		_id: 0,
		title: "Watch"
	},
	{
		_id: 1,
		title: "Clock"
	}
];

const item = renderer.create(
	<Item item={mockItemList[0]} onClick={() => {}} />
);
const productList = renderer.create(<ProductsList products={mockItemList} />);

describe("Products list", () => {
	it("item renders correctly", () => {
		expect(item).toMatchSnapshot();
	});

	it("list renders correctly", () => {
		expect(productList).toMatchSnapshot();
	});
});
