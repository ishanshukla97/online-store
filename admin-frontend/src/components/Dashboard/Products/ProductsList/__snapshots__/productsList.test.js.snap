// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products list item renders correctly 1`] = `
<a
  className="collection-item"
  href="#!"
  onClick={[Function]}
>
  Watch
</a>
`;

exports[`Products list list renders correctly 1`] = `
<div
  className="collection col s9"
  style={
    Object {
      "marginLeft": "5%",
    }
  }
>
  <a
    className="collection-item"
    href="#!"
    onClick={[Function]}
  >
    Watch
  </a>
  <a
    className="collection-item"
    href="#!"
    onClick={[Function]}
  >
    Clock
  </a>
</div>
`;

exports[`item renders correctly 1`] = `
<a
  className="collection-item"
  href="#!"
  onClick={[Function]}
>
  Watch
</a>
`;

exports[`list renders correctly 1`] = `
<div
  className="collection col s9"
  style={
    Object {
      "marginLeft": "5%",
    }
  }
>
  <a
    className="collection-item"
    href="#!"
    onClick={[Function]}
  >
    Watch
  </a>
  <a
    className="collection-item"
    href="#!"
    onClick={[Function]}
  >
    Clock
  </a>
</div>
`;
